src/core/evaluator.js

```diff
@@ -1640,19 +1647,38 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
                 xObjStateManager.transform(matrix);
               }
 
+              // Enqueue the `textContent` chunk before parsing the /Form
+              // XObject.
+              enqueueChunk();
+              let sinkWrapper = {
+                enqueueInvoked: false,
+                enqueue(chunk, size) {
+                  this.enqueueInvoked = true;
+                  sink.enqueue(chunk, size);
+                },
+
+                get desiredSize() {
+                  return sink.desiredSize;
+                },
+
+                get ready() {
+                  return sink.ready;
+                },
+              };
+
               next(self.getTextContent({
                 stream: xobj,
                 task,
                 resources: xobj.dict.get('Resources') || resources,
                 stateManager: xObjStateManager,
                 normalizeWhitespace,
                 combineTextItems,
-              }).then(function (formTextContent) {
-                Util.appendToArray(textContent.items, formTextContent.items);
-                Util.extendObj(textContent.styles, formTextContent.styles);
-
-                xobjsCache.key = name;
-                xobjsCache.texts = formTextContent;
+                sink: sinkWrapper,
+                seenStyles,
+              }).then(function () {
+                if (sinkWrapper.enqueueInvoked === false) {
+                  skipNonFormXobjs[name] = true;
```

Comment:
Similar to https://github.com/mozilla/pdf.js/pull/8488/commits/1e9429e8f925ad74ce119a7ec503f5ae6d434539#r120729155, this line needs to be removed, since you're not suppose to cache these XObjects any more.